hardware/raspberrypi/frequency-management.md

```diff
@@ -8,14 +8,27 @@ For Raspberry Pi 3 Model B+, the PCB technology has been changed to provide bett
 
 The Raspberry Pi 4 Model B continues with the same PCB technology as the Raspberry Pi 3B+ to help dissipate excess heat. There is currently no soft limit defined.
 
-### DVFS on the Raspberry Pi 4B
+### DVFS on the Raspberry Pi 4B, 400 and CM4
 
-On Raspberry Pi 4 Model B, firmware from late November 2019 onwards implements Dynamic Voltage and Frequency Scaling. This technique (outlined on Wikipedia [here](https://en.wikipedia.org/wiki/Dynamic_voltage_scaling)) allows Raspberry Pi 4B to run at lower temperatures whilst still providing the same performance.
+On Raspberry Pi 4 Model B, firmware from late November 2019 onwards implements Dynamic Voltage and Frequency Scaling (DVFS). This technique (outlined on Wikipedia [here](https://en.wikipedia.org/wiki/Dynamic_voltage_scaling)) allows Raspberry Pi 4B to run at lower temperatures whilst still providing the same performance.
 
 Various clocks (e.g. ARM, Core, V3D, ISP, H264, HEVC) inside the SoC are monitored by the firmware, and whenever they are not running at full speed, the voltage supplied to the particular part of the chip driven by the clock is reduced relative to the reduction from full speed. In effect, only enough voltage is supplied to keep the block running correctly at the specific speed at which it is running. This can result in significant reductions in power used by the SoC, and therefore in the overall heat being produced.
 
+Due to possible system stability problems involved with running an undervoltage, especially when using undervoltaged fixed clock peripherals (eg. PCIe), two DVFS modes are available and can be configured in `/boot/config.txt` with the below properties. GUI desktop systems should use `dvfs=2`, headless systems may benefit from `dvfs=1` provided peripheral use is managed.
+
+| property=value | Description         |
+|----------------|---------------------|
+| dvfs=1         | allow under-voltage |
+| dvfs=2         | no under-voltage    |
+
 In addition, a more stepped CPU governor is also used to produce finer-grained control of ARM core frequencies, which means the DVFS is more effective. The steps are now 1500MHz, 1000MHz, 750MHz, and 600MHz. These steps can also help when the SoC is being throttled, and mean that throttling all the way back to 600MHz is much less likely, giving an overall increase in fully loaded performance.
 
+The default CPU governor is `ondemand`, the governer can be manually changed for example with `sudo cpufreq-set -g powersave` to reduce idle power consumption.
+
+DVFS is supported on Raspberry Pi 4B, 400 and CM4 devices with supporting firmware.
```

Comment:
Doesn't this last sentence duplicate the "On Raspberry Pi 4 Model B, firmware from late November 2019 onwards implements Dynamic Voltage and Frequency Scaling (DVFS)." that was written earlier?

Reply:
I myself am developing for the Raspberry Pi CM4 so my initial read of the article made me think that maybe DVFS was for Raspberry Pi 4B only and not yet implemented on the 400 or CM4 so I wanted to be extra clear in that last sentence that they were supported (not sure when supported was added or if they were always supported so I didn't change the first sentence).

Reply:
I suggest changing the heading to "DVFS on the Raspberry Pi 4 (4B, 400 and CM4)", then removing all references to the individual models except where a differentiation is required (which I think is nowhere). November 2019 is sufficiently long ago (we've had most of a global pandemic since then) that I don't think it adds anything to mention when the feature was added.